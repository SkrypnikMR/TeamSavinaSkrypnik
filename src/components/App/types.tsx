import { colors } from '../UI/baseLayout';

export interface ITheme {
    theme: 'dark' | 'light';
    colors: typeof colors;
}

export interface IDiv extends ITheme {
    minHeight: string;
    background: string;
    backgroundSize: string;
    direction: string;
}
