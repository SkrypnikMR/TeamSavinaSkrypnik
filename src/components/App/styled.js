import styled from 'styled-components';

export const StAppDiv = styled.div`
  min-height: 100vh;
  ${({ url = 'heh' }) => url && `background: url(${url}) 100% 100% no-repeat`};
  background-size: cover;
  direction: ${({ lang }) => (lang === 'ar' ? 'rtl' : 'ltr')};
`;
