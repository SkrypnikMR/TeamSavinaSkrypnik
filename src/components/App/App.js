import React from 'react';
import { Route, Switch, BrowserRouter } from 'react-router-dom';
import PropTypes from 'prop-types';
import { NotificationContainer } from 'react-notifications';
import 'react-notifications/lib/notifications.css';
import { useTranslation } from 'react-i18next';
import Registration from '../Registration';
import Login from '../Login';
import Chat from '../Chat';
import MyAccount from '../MyAccount';
import ModalComponent from '../UI/Modal';
import Component from '../UI/Modal/ModalTestComponent.jsx';
import Header from '../Header';
import { APP_ROUTES } from '/src/constants/reactRoutes';
import { StAppDiv } from './styled';
import ModalInviteUsers from '../UI/Modals/ModalInviteUsers';
import ModalCreateRoom from '../UI/Modals/ModalCreateRoom';

const App = ({ userTheme, userThemeMode }) => {
  const { i18n } = useTranslation();
  return (
    <StAppDiv url={userThemeMode === 'dark' ? userTheme.dark : userTheme.light} lang={i18n.language}>
      <BrowserRouter>
        <Header />
        <Switch>
          <Route path={APP_ROUTES.login} exact component={Login} />
          <Route path={APP_ROUTES.registration} exact component={Registration} />
          <Route path={APP_ROUTES.chat} exact component={Chat} />
          <Route path={APP_ROUTES.account} exact component={MyAccount} />
        </Switch>
      </BrowserRouter>
      <ModalComponent
        Component={Component}
        headerTextKey="notification_settings"
      />
      <ModalCreateRoom />
      <ModalInviteUsers />
      <NotificationContainer />
    </StAppDiv >
  );
};

App.propTypes = {
  userTheme: PropTypes.object,
  userThemeMode: PropTypes.string,
};

export default App;
