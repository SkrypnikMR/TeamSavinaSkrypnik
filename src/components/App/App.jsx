import React from 'react';
import { Route, Switch, BrowserRouter } from 'react-router-dom';
import { NotificationContainer } from 'react-notifications';
import 'react-notifications/lib/notifications.css';
import { useTranslation } from 'react-i18next';
import { APP_ROUTES } from '../../constants/reactRoutes';
import Registration from '../Registration';
import Login from '../Login';
import ModalComponent from '../UI/Modal';
import Component from '../UI/Modal/ModalTestComponent.jsx';
import { StAppDiv } from './styled';
import MainPage from '../MainPage';

const App = () => {
  const { i18n } = useTranslation();
  return (
    <StAppDiv lang={i18n.language}>
      <BrowserRouter>
        <Switch>
          <Route path={APP_ROUTES.login} exact component={Login} />
          <Route path={APP_ROUTES.registration} exact component={Registration} />
          <Route path={APP_ROUTES.mainPage} exact component={MainPage} />
        </Switch>
      </BrowserRouter>
      <ModalComponent
        Component={Component}
        headerTextKey="notification_settings"
      />
      <NotificationContainer />
    </StAppDiv >
  );
};

export default App;
