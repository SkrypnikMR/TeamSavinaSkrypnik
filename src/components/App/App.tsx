import React from 'react';
import { Route, Switch, BrowserRouter } from 'react-router-dom';
import { NotificationContainer } from 'react-notifications';
import 'react-notifications/lib/notifications.css';
import { useTranslation } from 'react-i18next';

import { APP_ROUTES } from '../../constants/reactRoutes';
import Registration from '../Registration/Registration';
import Login from '../Login';
import MainPage from '../MainPage';
import StatisticPage from '../StatisticPage';
import { useTheme } from '../Hook/useTheme';

import { StAppDiv } from './styled';

const App = () => {
  const { i18n } = useTranslation();
  const { colors, theme } = useTheme();
  return (
    <StAppDiv lang={i18n.language} colors={colors} theme={theme}>
      <BrowserRouter>
        <Switch>
          <Route path={APP_ROUTES.login} exact component={Login} />
          <Route path={APP_ROUTES.registration} exact component={Registration} />
          <Route path={APP_ROUTES.mainPage} exact component={MainPage} />
          <Route path={APP_ROUTES.statistics} exact component={StatisticPage} />
        </Switch>
      </BrowserRouter>
      <NotificationContainer />
    </StAppDiv >
  );
};

export default App;
