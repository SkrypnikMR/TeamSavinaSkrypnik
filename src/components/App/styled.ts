import styled from 'styled-components';

import { IDiv } from './types';

export const StAppDiv = styled.div < IDiv > `
  min-height: 100vh;
  background: ${({ colors, theme }) => colors[theme].background};
  background-size: cover;
`;
