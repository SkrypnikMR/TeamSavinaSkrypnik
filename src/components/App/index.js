import { connect } from 'react-redux';
import { userTheme, userThemeMode } from 'src/Store/user/selectors';
import App from './App';

const mapStateToProps = state => ({
  userTheme: userTheme(state),
  userThemeMode: userThemeMode(state),
});

export default connect(mapStateToProps)(App);
