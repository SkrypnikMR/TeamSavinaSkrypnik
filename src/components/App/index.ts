import App from './App';
import { withTheme } from '../HOC';

export default withTheme(App);
