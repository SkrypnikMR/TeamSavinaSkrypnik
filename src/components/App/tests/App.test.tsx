import React from 'react';
import configureStore from 'redux-mock-store';
import { shallowSmart, mountSmart } from '../../../../__tests__/testHelper';
import App from '../App';

jest.mock('../../../index', () => ({ store: { dispatch: jest.fn() } }));

const mockStore = configureStore();
const store = mockStore({
  registration: {
    login: '',
    password: '',
    confirm: '',
    success: null,
  },
  login: {
    login: '',
    password: '',
    success: null,
  },
  game: {
     rooms: [],
    userLogin: localStorage.getItem('login') || '',
    actualRoom: {
      gameType: '',
      creatorLogin: '',
      guestLogin: '',
      startTime: 0,
      id: '',
      stepDoList: [],
    },
    stepOrder: '',
    stepHistory: [],
    winner: '',
    possibleSteps: [],
  },
  statistic: {
    fullStatistic: [],
  },
});

describe('App', () => {
  it('Should match snapshot', () => {
    const component = shallowSmart(<App />, store);
    expect(component.html()).toMatchSnapshot();
  });
  it('Should render App', () => {
    const component = mountSmart(<App />, store);
    expect(component.html()).toMatchSnapshot();
  });
});
