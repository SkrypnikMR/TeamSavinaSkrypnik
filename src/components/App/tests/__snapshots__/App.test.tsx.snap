// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Should match snapshot 1`] = `"<div class=\\"styled__StAppDiv-sc-181nq3d-0 knFUYF\\"><div class=\\"styled__StLoginDiv-sc-17zo54h-0 hHzyfF\\"><img src=\\"../../../public/assets/images/game-development.png\\"/><div class=\\"styled__StFormDiv-sc-p60jdb-0 kshiNm\\"><p>authorization</p><div width=\\"80%\\" height=\\"100px\\" class=\\"styled__StInputContainer-sc-e64755-3 bIHjkn\\"><label for=\\"login\\" class=\\"styled__StLabel-sc-e64755-0 kqGJQd\\">login_label</label><input type=\\"text\\" id=\\"login\\" name=\\"login\\" value=\\"\\" placeholder=\\"login_placeholder\\" class=\\"styled__StInput-sc-e64755-1 kmoiCN\\"/></div><div width=\\"80%\\" height=\\"100px\\" class=\\"styled__StInputContainer-sc-e64755-3 bIHjkn\\"><label for=\\"password\\" class=\\"styled__StLabel-sc-e64755-0 kqGJQd\\">password_label</label><input type=\\"text\\" id=\\"password\\" name=\\"password\\" value=\\"\\" placeholder=\\"password_placeholder\\" class=\\"styled__StInput-sc-e64755-1 kmoiCN\\"/></div><button id=\\"login\\" type=\\"button\\" height=\\"50px\\" class=\\"styled__StButton-sc-1y35zax-0 cHOOvi\\">login_btn</button><span>login_span_text<a href=\\"/registration\\">login_link_text</a></span></div></div><div class=\\"notification-container notification-container-empty\\"><div></div></div></div>"`;
